### DISCUSSION TITLE
Setting recalc_transform=False when saving aligned raster on disk

### DISCUSSION THREAD

**Discussion by victoriafoing** [corteva/rioxarray#659] [General]
https://corteva.github.io/rioxarray/stable/examples/reproject_match.html

I want to align a raster to another reference raster on the disk. When I use reproject_match to align a raster in memory, it has the same transform of the reference raster (e.g. bounds, crs, resolution). However, when I save the aligned raster with rio.to_raster(), it has a slightly different transform than the reference raster. Do I need to set recalc_transform=False when saving the aligned raster to make sure it is aligned with the reference raster?

**Comment by snowman2** [comment]
Can you provide a minimal re-producible example?

**Comment by victoriafoing** [comment]
If I align a raster to a reference raster (or even align it to itself) using the following:
```
import rioxarray
from rasterio.enums import Resampling

reference_raster = rioxarray.open_rasterio(reference_raster_path)
array_coords = {
    "x": reference_raster.x,
    "y": reference_raster.y
}
reprojected_raster = reference_raster.rio.reproject_match(
    reference_raster, resampling=Resampling.nearest
)
aligned_raster = reprojected_raster.assign_coords(array_coords)
```

And then I save the raster with recalc_transform=False, its resolution matches that of the reference raster.
```
aligned_raster.rio.to_raster(aligned_raster_path, recalc_transform=False)
aligned_raster = rioxarray.open_rasterio(aligned_raster_path)
aligned_raster.rio.resolution () == reference_raster.rio.resolution()
# Returns True
```

But if I save the raster with recalc_transform=True, its resolution does not match that of the reference raster.
```
aligned_raster.rio.to_raster(aligned_raster_path)
aligned_raster = rioxarray.open_rasterio(aligned_raster_path)
aligned_raster.rio.resolution () == reference_raster.rio.resolution()
# Returns False
```
I want the saved aligned raster to have the same bounds, resolution, and coordinates as the reference raster. Do I always have to set recalc_transform=False?

**Reply by snowman2** [reply]
Side note: You don't need to do `assign_coords` as `reproject_match` does that for you already.

What happens if you write the transform from the original raster to the new raster using `rio.write_transform`? [documentation](https://corteva.github.io/rioxarray/stable/getting_started/crs_management.html#Setting-the-transform-of-the-dataset)
